RippleCarryAdder/ReferenceImplementation.qs

```diff
@@ -73,28 +73,24 @@ namespace Quantum.Kata.RippleCarryAdder {
     // Task 1.7. N-bit adder
     operation ArbitraryAdder_Reference (a : Qubit[], b : Qubit[], sum : Qubit[], carry : Qubit) : Unit is Adj {
         let N = Length(a);
-        if (N == 1) {
-            LowestBitSum_Reference(a[0], b[0], sum[0]);
-            LowestBitCarry_Reference(a[0], b[0], carry);
+        if N == 1 {
+            OneBitAdder(a[0], b[0], sum[0], carry);
         }
         else {
-            use internalCarries = Qubit[N-1];
-            LowestBitSum_Reference(a[0], b[0], sum[0]);
-            LowestBitCarry_Reference(a[0], b[0], internalCarries[0]);
-                
-            for i in 1 .. N-2 {
-                HighBitSum_Reference(a[i], b[i], internalCarries[i-1], sum[i]);
-                HighBitCarry_Reference(a[i], b[i], internalCarries[i-1], internalCarries[i]);
+            use carries = Qubit[N-1];
```

Comment:
I would keep the name `internalCarries`, to make it easier to distinguish it from the `carry` parameter. This will also help GitHub identify the differences between the old code and the new one easier